src/olympia/devhub/views.py

```diff
@@ -2016,66 +2015,89 @@ def api_key(request):
             % (reverse('devhub.developer_agreement'), '?to=', quote(request.path))
         )
 
-    try:
-        credentials = APIKey.get_jwt_key(user=request.user)
-    except APIKey.DoesNotExist:
-        credentials = None
+    form = forms.APIKeyForm(
+        request.POST if request.method == 'POST' else None,
+        user=request.user,
+        action=request.POST.get('action') or None,
+    )
 
-    try:
-        confirmation = APIKeyConfirmation.objects.get(user=request.user)
-    except APIKeyConfirmation.DoesNotExist:
-        confirmation = None
+    if request.method == 'POST' and form.is_valid():
+        # user is trying to revoke/regenerate credentials, clear existing credentials
+        if form.action in (form.ACTION_CHOICES.revoke, form.ACTION_CHOICES.regenerate):
+            form.revoke_credentials()
+            log.info(
+                f'revoking JWT key for user: {request.user.id}, {form.credentials}'
+            )
+            send_key_revoked_email(request.user.email, form.credentials.key)
 
-    if request.method == 'POST':
-        has_confirmed_or_is_confirming = confirmation and (
-            confirmation.confirmed_once
-            or confirmation.is_token_valid(request.POST.get('confirmation_token'))
-        )
+            if form.action == form.ACTION_CHOICES.revoke:
+                form.confirmation.delete()
```

Comment:
this is the biggest departure from the previous flow.. but I think this is a good additional feature. If you explicitly revoke your own keys we should probably double check you're human and the account owner... otherwise you end up on a screen waiting for a confirmation token that might be really old and hard to find.